src/eva/eva_driver.py

```diff
@@ -9,16 +9,170 @@
+
+    Parameters:
+        logger (Logger): An instance of the logger for logging messages.
+        timing (Timing): An instance of the timing object for timing the process.
+        eva_dict (dict): The configuration dictionary for the EVA process.
+        data_collections (DataCollections): An instance of the data collections object.
+
+    Returns:
+        None
+    """
+
+    # Get the datasets configuration
+    datasets_config = get(eva_dict, logger, 'datasets')
+
+    # Optionally suppress the display of the collection
+    suppress_collection_display = get(eva_dict, logger, 'suppress_collection_display', False)
+
+    # Loop over datasets reading each one in turn, internally appending the data_collections
+    for dataset_config in datasets_config:
+
+        # Prepare diagnostic data
+        logger.info('Running data driver')
+        timing.start('DataDriverExecute')
+        data_driver(dataset_config, data_collections, timing, logger)
+        timing.stop('DataDriverExecute')
+
+    # After reading all datasets display the collection
+    if not suppress_collection_display:
+        logger.info('Reading of Eva data complete: status of collections: ')
+        data_collections.display_collections()
+
+    # Perform any transforms
+    if 'transforms' in eva_dict:
+        logger.info(f'Running transform driver')
+        timing.start('TransformDriverExecute')
+        transform_driver(eva_dict, data_collections, timing, logger)
+        timing.stop('TransformDriverExecute')
+
+        # After reading all datasets display the collection
+        if not suppress_collection_display:
+            logger.info('Transformations of data complete: status of collections: ')
+            data_collections.display_collections()
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+def read_transform_time_series(logger, timing, eva_dict, data_collections):
+
+    """
+    Read the data and perform transforms on the fly. Then collapse data into time series
+
+    Parameters:
+        logger (Logger): An instance of the logger for logging messages.
+        timing (Timing): An instance of the timing object for timing the process.
+        eva_dict (dict): The configuration dictionary for the EVA process.
+        data_collections (DataCollections): An instance of the data collections object.
+
+    Returns:
+        None
+    """
+
+    # Check for required keys
+    # -----------------------
+    required_keys = [
+        'begin_date',
+        'final_date',
+        'interval',
+        'collection',
+        'variables',
+        ]
+    for key in required_keys:
+        logger.assert_abort(key in eva_dict['time_series'], 'If running eva in time series ' +
+                            f'mode the time series config must contain "{key}"')
+
+    # Write message that this is a time series run
+    logger.info('This instance of Eva is being used to accumulate a time series.')
```

Comment:
random, but above line is `eva`, this line is `Eva`, we should be consistent. Let's call it "technical debt", and assign @guillaumevernieres to fix it all for us :-)